product_code/matriculation_service/impl/src/main/scala/de/upb/cs/uc4/matriculation/impl/MatriculationServiceImpl.scala

```diff
@@ -41,7 +41,16 @@ class MatriculationServiceImpl(clusterSharding: ClusterSharding, userService: Us
     authenticated[PutMessageMatriculation, Done](AuthenticationRole.Admin) {
       ServerServiceCall { (header, rawMessage) =>
         val message = rawMessage.trim
-        val validationList = message.validate
+
+        // For syntax and regex checks, 5 seconds are more than enough
+        val validationList = try {
+          Await.result(message.validate, 5.seconds)
```

Comment:
The timeout should be defined in the application.conf
And than loaded into a lazy val